examples/feature_engineering.py

```diff
@@ -0,0 +1,333 @@
+
+
+@task
+def get(values: List[np.ndarray], keys: List[str], key: str) -> np.ndarray:
+    """
+    Retrieve the value associated with a key given a list of keys and list of values.
+
+    Essentially a utility to make use of dicts that have been broken into
+    a list of keys and a list of values.
+
+    Args:
+        - values (List[np.ndarray]): values associated with keys
+        - keys (List[str]): keys to choose from
+        - key (str): the chosen key
+
+    Returns:
+        - np.ndarray: the value associated with the chosen key
+    """
+    return values[keys.index(key)]
+
+
+# using a specific number of arrays since Prefect doesn't support variable positional args
+@task
+def concatenate4(
+    a: np.ndarray, b: np.ndarray, c: np.ndarray, d: np.ndarray
+) -> np.ndarray:
+    """
+    Concatenate 4 arrays along the last axis.
+    for 1D arrays, make a 2D array of columns.
+    """
+    arrays = [a, b, c, d]
+    for i, a in enumerate(arrays):
+        if len(a.shape) == 1:
+            arrays[i] = np.expand_dims(a, -1)
+    return np.concatenate(arrays, axis=-1)
+
+
+@task
+def concatenate8(
+    a: np.ndarray,
+    b: np.ndarray,
+    c: np.ndarray,
+    d: np.ndarray,
+    e: np.ndarray,
+    f: np.ndarray,
+    g: np.ndarray,
+    h: np.ndarray,
+) -> np.ndarray:
+    """
+    Concatenate 8 arrays along the last axis.
+    for 1D arrays, make a 2D array of columns.
+    """
+    arrays = [a, b, c, d, e, f, g, h]
+    for i, a in enumerate(arrays):
+        if len(a.shape) == 1:
+            arrays[i] = np.expand_dims(a, -1)
+    return np.concatenate(arrays, axis=-1)
+
+
+@task
+def dot(A: np.ndarray, B: np.ndarray) -> np.ndarray:
+    """
+    Calculate the dot product of two arrays.
+    """
+    return np.dot(A, B)
+
+
+# sabermetric calculation tasks
+@task
+def babip(
+    h: np.ndarray, hr: np.ndarray, ab: np.ndarray, k: np.ndarray, sf: np.ndarray
+) -> np.ndarray:
+    """Calculate Batting Average on Balls In Play statistic."""
+    X1 = h - hr
+    X2 = ab - k - hr + sf
+    impute_array = np.zeros_like(X1)
+    return np.divide(
+        X1.astype(np.float16), X2, out=impute_array.astype(np.float16), where=X2 != 0
+    )
+
+
+@task
+def obp(
+    h: np.ndarray, bb: np.ndarray, hbp: np.ndarray, ab: np.ndarray, sf: np.ndarray
+) -> np.ndarray:
+    """Calculate On-Base Percentage statistic."""
+    X1 = h + bb + hbp
+    X2 = ab + bb + hbp + sf
+    impute_array = np.zeros_like(X1)
+    return np.divide(
+        X1.astype(np.float16), X2, out=impute_array.astype(np.float16), where=X2 != 0
+    )
+
+
+# utility
+class DataFrame:
+    """A utility class to provide convenient syntax for grabbing columns as a Task."""
+
+    def __init__(self, cols: "Task", colnames: "Task"):
+        self.cols = cols
+        self.colnames = colnames
+
+    def __getitem__(self, key: str):
+        return get(self.cols, self.colnames, key)
+
+
+# let's do some ML for fun
+class PCATask(Task):
+    def __init__(self, *args, **kwargs):
```

Comment:
I know you don't use any other `Task` initialization kwargs in this example, but in general I'd recommend allowing for optionally passing `Task` initialization kwargs to the `Task` init method.

Reply:
I wasn't 100% sure how to go about this, but is:
```
class PCATask(Task):
    def __init__(self, *args, task_kwargs={}, **kwargs):
```

the right approach here? Since as far as I know there isn't room for two **kwargs.

Reply:
Yea I think this is solid!